pcsx2/x86/ix86-32/iR5900-32.cpp

```diff
@@ -1651,15 +1651,33 @@ static void __fastcall recRecompile( const u32 startpc )
 
 	pxAssert(s_pCurBlockEx);
 
-	if (HWADDR(startpc) == EELOAD_START) {
-		// The EELOAD _start function is the same across all BIOS versions afaik
+	if (HWADDR(startpc) == EELOAD_START)
+	{
+		// The EELOAD _start function is the same across all BIOS versions
 		u32 mainjump = memRead32(EELOAD_START + 0x9c);
 		if (mainjump >> 26 == 3) // JAL
-			eeloadMain = ((EELOAD_START + 0xa0) & 0xf0000000U) | (mainjump << 2 & 0x0fffffffU);
+			g_eeloadMain = ((EELOAD_START + 0xa0) & 0xf0000000U) | (mainjump << 2 & 0x0fffffffU);
 	}
 
-	if (eeloadMain && HWADDR(startpc) == HWADDR(eeloadMain)) {
+	if (g_eeloadMain && HWADDR(startpc) == HWADDR(g_eeloadMain))
+	{
 		xFastCall((void*)eeloadHook);
+		if (g_SkipBiosHack)
+		{
+			// There are four known versions of EELOAD, identifiable by the location of the 'jal' to the EELOAD function which
+			// calls ExecPS2(). The function itself is at the same address in all BIOSs after v1.00-v1.10.
+			u32 typeAexecjump = memRead32(EELOAD_START + 0x470); // v1.00, v1.01?, v1.10?
+			u32 typeBexecjump = memRead32(EELOAD_START + 0x5B0); // v1.20, v1.50, v1.60 (3000x models)
+			u32 typeCexecjump = memRead32(EELOAD_START + 0x618); // v1.60 (3900x models)
+			u32 typeDexecjump = memRead32(EELOAD_START + 0x600); // v1.70, v1.90, v2.00, v2.20, v2.30
+			if ((typeBexecjump >> 26 == 3) || (typeCexecjump >> 26 == 3) || (typeDexecjump >> 26 == 3)) // JAL to 0x822B8
+				g_eeloadExec = EELOAD_START + 0x2B8;
+			else if (typeAexecjump >> 26 == 3) // JAL to 0x82170
+				g_eeloadExec = EELOAD_START + 0x170;
+			else // There might be other types of EELOAD, because these models' BIOSs have not been examined: 18000, 3500x, 3700x,
+				 // 5500x, and 7900x. However, all BIOS versions have been examined except for v1.01 and v1.10.
+				Console.WriteLn("recRecompile: Could not enable launch arguments for fast boot mode; unidentified BIOS version! Please report this to the PCSX2 developers.");
```

Comment:
Could you explain what happens in this case? 
It would be great if it just continued normally with fast boot.

Reply:
It will.  In this `else` case, g_eeloadExec will remain initialized at zero, so the `if (g_eeloadExec …)` check below will fail and eeloadHook2() will not be called.  I wanted the above code, however, to run on all fast boots as this will allow us to more quickly discover any EELOAD versions which have the ExecExecutable() call in a different place, at which points those EELOADs can be added to the list above.  Admittedly, most users will not notice or report the message in their console.

Reply:
That sounds good to me.
I don't expect many more official versions by the way. I just worried about customized or obscure development images that someone may try out.
In those cases, it's okay for this function to bail out and continue normally :)